README.md

```diff
@@ -1,10 +1,260 @@
-# adminrouter
-The admin router is an open-source Nginx configuration created by Mesosphere that provides central authentication and proxy to DCOS services within the cluster.
+# Admin Router
 
+The Admin Router is an open-source Nginx configuration created by
+Mesosphere that provides central authentication and proxy to DC/OS services
+within the cluster.
 
 <img src="admin-router.png" alt="" width="100%" align="middle">
 
-## Ports
+## Ports summary
 <img src="admin-router-table.png" alt="" width="100%" align="middle">
- 
- 
+
+## Testing
+
+Admin Router repository includes a test harness that is meant to make
+testing easier and in some cases - possible. It's written in Python and
+uses pytest fixtures and custom modules to mock out all relevant DC/OS
+features and control Nginx startup and termination.
+
+All the tests are executed in a Docker container which is controlled by the
+Makefile. Inside the container pytest command is started which in turn pulls
+in all the relevant fixtures, such as syslog mock, mocker (DC/OS endpoints
+mock), dns mock, etc... Finally, an Nginx is spawned using the configuration
+bind-mounted from the developer's repository. Tests may launch Nginx multiple
+times, in different configuration, depending on what is needed. After the
+tests runner finishes, all the processes and the environment is cleaned up
+by pytest.
+
+Below, there is a general overview of all the components of the test harness.
+More detailed documentation can be found in docstrings and comments in the code
+itself. If in doubt, it also may be very helpful to look for examples among
+existing tests.
+
+### Quickstart
+In order to execute all the test, just issue:
+
+	make test
+
+In order have fine grained control over pytest command, execute:
+
+	make shell
+
+This will launch an interactive environment inside the container that has all
+the dependencies installed. The developer may now launch the `pytest` command
+the way it's necessary, debug the environment and temporarily add/change
+the dependencies.
+
+### Makefile
+Makefile provides an easy way to start the testing environment without the need
+to worry about the correct docker commands. It's core concept is `adminrouter-devkit`
+container which is contains all the dependencies that are needed to run
+Admin Router, and inside which all tests-related commands are run.
+
+It exposes a couple of targets:
+* `make clean` - remove all containers created by the test harness. It does not
+  remove the images themselves though as the layer cache may be useful later
+  on and the user may remove them themselves.
```

Comment:
or 'and the user may remove them X'. (where X is one of himself/herself/itself/manually)